crypto/merkletrie/cache_test.go

```diff
@@ -117,3 +123,80 @@ func TestCacheEviction3(t *testing.T) {
 		}
 	}
 }
+
+// smallPageMemoryCommitter is an InMemoryCommitter, which has a custom page size, and knows how to "fail" per request.
+type smallPageMemoryCommitter struct {
+	InMemoryCommitter
+	pageSize  int64
+	failStore int
+	failLoad  int
+}
+
+// GetNodesCountPerPage returns the page size ( number of nodes per page )
+func (spmc *smallPageMemoryCommitter) GetNodesCountPerPage() (pageSize int64) {
+	return spmc.pageSize
+}
+
+// StorePage stores a single page in an in-memory persistence.
+func (spmc *smallPageMemoryCommitter) StorePage(page uint64, content []byte) error {
+	if spmc.failStore > 0 {
+		spmc.failStore--
+		return fmt.Errorf("failStore>0")
+	}
+	return spmc.InMemoryCommitter.StorePage(page, content)
+}
+
+// LoadPage load a single page from an in-memory persistence.
+func (spmc *smallPageMemoryCommitter) LoadPage(page uint64) (content []byte, err error) {
+	if spmc.failLoad > 0 {
+		spmc.failLoad--
+		return nil, fmt.Errorf("failLoad>0")
+	}
+	return spmc.InMemoryCommitter.LoadPage(page)
+}
+
+// TestCacheEvictionFuzzer generates bursts of random Add/Delete operations on the trie, and
+// testing the correctness of the cache internal buffers priodically.
+func TestCacheEvictionFuzzer(t *testing.T) {
+	// create 2000 hashes.
+	leafsCount := 2000
+	hashes := make([]crypto.Digest, leafsCount)
+	for i := 0; i < len(hashes); i++ {
+		hashes[i] = crypto.Hash([]byte{byte(i % 256), byte((i / 256) % 256), byte(i / 65536)})
+	}
+	for _, pageSize := range []int64{2, 3, 8, 12, 17} {
```

Comment:
make page size and evict size random in a range, it would make more sense

Reply:
I was hoping to have some controlled dimensionality. I don't mind having some pure randomized version of these as well.

Reply:
yeah, generate some additional values and add to iterations